modules/oas-page-builder/src/services/pageBuilder.ts

```diff
@@ -0,0 +1,82 @@
+import fetch from 'node-fetch';
+import { ModuleOptions } from '../types';
+import { normalizePath } from '../utils/normalizePath';
+import { execRedoc } from './commandExecutor';
+import { findLastSavedGitHash } from './database';
+import { OASPageMetadata } from './types';
+
+const fetchTextData = async (url: string, errMsg: string) => {
+  const res = await fetch(url);
+  if (!res.ok) {
+    // Error should be caught when creating pages.
+    throw new Error(`${errMsg}; ${res.statusText}`);
+  }
+  return res.text();
+};
+
+const getOASFileUrl = (gitHash: string) =>
+  `https://mongodb-mms-prod-build-server.s3.amazonaws.com/openapi/${gitHash}.json`;
+
+const getAtlasSpecUrl = async (apiKeyword: string) => {
+  // Currently, the only expected API fetched programmatically is the Cloud Admin API,
+  // but it's possible to have more in the future with varying processes.
+  if (apiKeyword !== 'cloud') {
+    throw new Error(`${apiKeyword} is not a supported API for building.`);
+  }
```

Comment:
minor nitpick: if we want more than one keyword, may be useful to follow the
````
const keywords = ['cloud'];
if keywords.includes(apiKeyword)
```

pattern